File: test_po.py
import urllib.request
import json
url = 'https://script.google.com/macros/s/AKfycbx0vjv3WWF9GXxA2zS16kNkLR0Nn9sdIfSUTIB7smfQbPhLRSX27P78OVFBJEPM9nJ9TQ/exec'
h_res = json.loads(urllib.request.urlopen(url + '?action=getHistory').read().decode('utf-8'))
r_res = json.loads(urllib.request.urlopen(url + '?action=getReceivedPOs').read().decode('utf-8'))
received = r_res.get('data', [])
for o in h_res.get('data', []):
    if o.get('orderNo') in received:
        print(f"PO {o.get('orderNo')} ({o.get('poMonth')}) is received")

File: test_po.js
const https = require('https');

const url = 'https://script.google.com/macros/s/AKfycbx0vjv3WWF9GXxA2zS16kNkLR0Nn9sdIfSUTIB7smfQbPhLRSX27P78OVFBJEPM9nJ9TQ/exec';

function fetchUrl(targetUrl) {
  return new Promise((resolve, reject) => {
    https.get(targetUrl, (res) => {
      // Handle redirect
      if (res.statusCode >= 300 && res.statusCode < 400 && res.headers.location) {
        return resolve(fetchUrl(res.headers.location));
      }
      
      let data = '';
      res.on('data', chunk => data += chunk);
      res.on('end', () => {
        try {
          resolve(JSON.parse(data));
        } catch(e) {
          reject(new Error('Failed to parse JSON: ' + data));
        }
      });
    }).on('error', reject);
  });
}

async function test() {
  try {
    const r_res = await fetchUrl(url + '?action=getReceivedPOs');
    console.log(r_res);
  } catch (e) {
    console.error(e);
  }
}

test();
